File: app/src/main/java/alireza/nezami/dependency_injection_without_dagger/java/approach3/MyApplication.java
package alireza.nezami.dependency_injection_without_dagger.java.approach3;

import android.app.Application;

import alireza.nezami.dependency_injection_without_dagger.java.approach3.data.local.DatabaseService;
import alireza.nezami.dependency_injection_without_dagger.java.approach3.data.remote.NetworkService;
import alireza.nezami.dependency_injection_without_dagger.java.approach3.di.DependencyComponent;

public class MyApplication extends Application {

    public NetworkService networkService;
    public DatabaseService databaseService;

    @Override
    public void onCreate() {
        super.onCreate();
        DependencyComponent.inject(this);
    }
}

File: app/src/main/java/alireza/nezami/dependency_injection_without_dagger/kotlin/approach3/MyApplication.kt
package alireza.nezami.dependency_injection_without_dagger.kotlin.approach3

import alireza.nezami.dependency_injection_without_dagger.kotlin.approach3.data.local.DatabaseService
import alireza.nezami.dependency_injection_without_dagger.kotlin.approach3.data.remote.NetworkService
import alireza.nezami.dependency_injection_without_dagger.kotlin.approach3.di.DependencyComponent
import android.app.Application

class MyApplication : Application() {
    var networkService: NetworkService? = null
    var databaseService: DatabaseService? = null
    override fun onCreate() {
        super.onCreate()
        DependencyComponent.inject(this)
    }
}